packages/snaps-controllers/src/snaps/registry/json.ts

```diff
@@ -153,6 +156,21 @@ export class JsonSnapsRegistry extends BaseController<
     );
   }
 
+  async #triggerUpdate() {
+    // If an update is ongoing, wait for that.
+    if (this.#currentUpdate) {
+      await this.#currentUpdate;
```

Comment:
Why is it needed to `await` here when it's already been awaited on line `169` initially (on the first call)?

Reply:
I'm not sure I follow?

If an update already exists and we call `triggerUpdate` we await the existing promise and return after that. If no update currently exists we create a promise and await it.

Reply:
I understand the case now. All good 👍